examples/actix-example.rs

```diff
@@ -0,0 +1,148 @@
+use actix_web::{web, App, HttpResponse, HttpServer, Responder};
+use eventsource_client as es;
+use futures::channel::mpsc::channel;
+use futures::TryStreamExt;
+use serde::{self, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::time::Duration;
+
+#[derive(Serialize)]
+struct Status {
+    capabilities: Vec<String>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct Config {
+    /// The URL of the SSE endpoint created by the test harness.
+    url: String,
+    /// A string describing the current test, if desired for logging.
+    tag: Option<String>,
+    /// An optional integer specifying the initial reconnection delay parameter, in
+    /// milliseconds. Not all SSE client implementations allow this to be configured, but the
+    /// test harness will send a value anyway in an attempt to avoid having reconnection tests
+    /// run unnecessarily slowly.
+    initial_delay_ms: Option<u64>,
+    /// An optional string which should be sent as the Last-Event-Id header in the initial
+    /// HTTP request. The test harness will only set this property if the test service has the
+    /// "last-event-id" capability.
+    last_event_id: Option<String>,
+    /// A JSON object containing additional HTTP header names and string values. The SSE
+    /// client should be configured to add these headers to its HTTP requests; the test harness
+    /// will then verify that it receives those headers. The test harness will only set this
+    /// property if the test service has the "headers" capability. Header names can be assumed
+    /// to all be lowercase.
+    headers: Option<HashMap<String, String>>,
+    /// A string specifying an HTTP method to use instead of GET. The test harness will only
+    /// set this property if the test service has the "post" or "report" capability.
+    method: Option<String>,
+    /// A string specifying data to be sent in the HTTP request body. The test harness will
+    /// only set this property if the test service has the "post" or "report" capability.
+    body: Option<String>,
+}
+
+#[derive(Serialize, Debug)]
+#[serde(tag = "kind")]
+enum EventType<'a> {
+    #[serde(rename = "hello")]
+    Hello,
+    #[serde(rename = "event")]
+    Event { event: Event<'a> },
+    #[serde(rename = "comment")]
+    Comment { comment: String },
+    #[serde(rename = "error")]
+    Error { error: String },
+}
+
+#[derive(Serialize, Debug)]
+struct Event<'a> {
+    #[serde(rename = "type")]
+    event_type: String,
+    data: Option<String>,
+    id: Option<&'a [u8]>,
+}
+
+async fn status() -> impl Responder {
+    web::Json(Status {
+        capabilities: vec![
+            // "headers".into(),
```

Comment:
Why is this commented out?

Reply:
Because we aren't supporting that capability yet. This is something we need to add support for still. It gets removed in a subsequent commit.